controllers/authentication.go

```diff
@@ -6,43 +6,8 @@ import (
 	"errors"
 	"io/ioutil"
 	"net/http"
-
-	"github.com/gin-gonic/contrib/sessions"
-	"github.com/gin-gonic/gin"
-	"github.com/thoas/go-funk"
 )
 
-func AuthenticationRequired(auths ...string) gin.HandlerFunc {
-	return func(c *gin.Context) {
-		session := sessions.Default(c)
-		user := session.Get("user")
-		id_token := c.GetHeader("id_token") // not sure what we'll call the key, id_token seems fine
-		if user == nil {
-			c.JSON(http.StatusUnauthorized, gin.H{"error": "user needs to be signed in to access this service"})
-			c.Abort()
-			return
-		}
-		if len(auths) != 0 {
-			authType := session.Get("authType")
-			if authType == nil || !funk.ContainsString(auths, authType.(string)) {
-				c.JSON(http.StatusForbidden, gin.H{"error": "invalid request, restricted endpoint"})
-				c.Abort()
-				return
-			}
-		}
-		// add session verification here, like checking if the user and authType
-		// combination actually exists if necessary. Try adding caching this (redis)
-		// since this middleware might be called a lot
-		_, err := authenticate(id_token)
-		if err != nil {
-			c.JSON(http.StatusUnauthorized, gin.H{"error": "could not authenticate id token -- " + err.Error()})
-			c.Abort()
-			return
-		}
-		c.Next()
-	}
-}
-
 // returns [google id], [error message] if the token is valid
 func authenticate(id_token string) (string, error) {
```

Comment:
This function should check if the id token is already in the cache. Ideally, this function would be an authentication black box that just says whether the token is valid, whether this means a cache miss followed by a call to the Google API, or a cache hit.
Furthermore, I think we can ignore the google id for now and simplify the signature of this function to be something like `func authenticate(idToken string) error`
If we start using the id for something in the future, then we can keep it in the cache, but for now it would make this function simpler to ignore the id.